packages/client/hmi-client/src/services/knowledge.ts

```diff
@@ -36,21 +36,35 @@ export async function fetchExtraction(id: string) {
 /**
  * Transform a list of LaTeX strings to an AMR
  * @param latex string[] - list of LaTeX strings representing a model
+ * @param modelId string - the model id to use for the extraction
  * @param framework [string] - the framework to use for the extraction, default to 'petrinet'
  * @return {Promise<Model | null>}
  */
-const latexToAMR = async (latex: string[], framework = 'petrinet'): Promise<Model | null> => {
+const latexToAMR = async (
+	latex: string[],
+	modelId?: string,
+	framework = 'petrinet'
+): Promise<Model | null> => {
```

Comment:
These parameters don't make a lot of sense, you don't want to have an optional parameter in the middle because you can never specify

```
latexToAMR(blah, 'ABM_framework')
```

Reply:
from what I understand `modelId` and `framework` are both optional, no? `framework` just happens to have a default value. For example, if you look at the calls to `latexToAMR` in either `tera-model-equation.vue` or `tera-model-diagram.vue`, you'll see that the calls are just `latexToAMR(equations, model id)`

Reply:
Judging by the way the API looks

```
`/knowledge/latex-to-amr/${framework}?modelId=${modelId}`
```

- framework is not optional, it just happens to have a default value, the value must be provided for the API end point to work
- modelId is optional, as it isn't needed for the API

These are not the same type of "optional" behaviour. The way you wrote up the interface makes it impossible to use a different framework.

Reply:
I don't think we use latexToAMR much (only in one or two places?). So it might be easier just to be explicit and do away with these kind of optional paradigms.